sdv/constraints/tabular.py

```diff
@@ -254,7 +254,7 @@ def _validate_metadata_specific_to_constraint(metadata, **kwargs):
         invalid_columns = []
         column_names = kwargs.get('column_names')
         for column in column_names:
-            if metadata.columns[column]['sdtype'] not in ['boolean', 'categorical']:
+            if metadata.get_columns()[column]['sdtype'] not in ['boolean', 'categorical']:
```

Comment:
This won't be backwards compatible. `SingleTableMetadata` does not have `get_columns()`.

Reply:
Made it backwards compatible by making sure all metadata is converted and also convert it the synthesizer metadata on `load`.

Also added an integration test for backward compatibility.